tasks/fevm/vm/task.go

```diff
@@ -0,0 +1,188 @@
+	// execute in parallel as both operations are slow
+	grp, _ := errgroup.WithContext(ctx)
+	var mex []*lens.MessageExecution
+	grp.Go(func() error {
+		var err error
+		mex, err = t.node.MessageExecutions(ctx, current, executed)
+		if err != nil {
+			return fmt.Errorf("getting messages executions for tipset: %w", err)
+		}
+		return nil
+	})
+
+	var getActorCode func(ctx context.Context, a address.Address) (cid.Cid, bool)
+	grp.Go(func() error {
+		var err error
+		getActorCode, err = util.MakeGetActorCodeFunc(ctx, t.node.Store(), current, executed)
+		if err != nil {
+			return fmt.Errorf("failed to make actor code query function: %w", err)
+		}
+		return nil
+	})
+
+	report := &visormodel.ProcessingReport{
+		Height:    int64(current.Height()),
+		StateRoot: current.ParentState().String(),
+	}
+
+	// if either fail, report error and bail
+	if err := grp.Wait(); err != nil {
+		report.ErrorsDetected = err
+		return nil, report, nil
+	}
+
+	var (
+		vmMessageResults = make(fevm.FEVMVMMessageList, 0)
+	)
+
+	for _, parentMsg := range mex {
+		// Only handle EVM related message
+		if !util.IsEVMAddress(ctx, t.node, parentMsg.Message.From, current.Key()) && !util.IsEVMAddress(ctx, t.node, parentMsg.Message.To, current.Key()) && !(parentMsg.Message.To != builtintypes.EthereumAddressManagerActorAddr) {
+			continue
+		}
+		messageHash, err := ethtypes.EthHashFromCid(parentMsg.Cid)
+		if err != nil {
+			log.Errorf("Error at finding hash: [cid: %v] err: %v", parentMsg.Cid, err)
+			continue
+		}
+		transaction, err := t.node.EthGetTransactionByHash(ctx, &messageHash)
+		if err != nil {
+			log.Errorf("Error at getting receipt: [hash: %v] err: %v", messageHash, err)
+			continue
+		}
+
+		if transaction == nil {
+			continue
+		}
+
+		log.Infof("message: %v, %v", parentMsg.Cid, base64.StdEncoding.EncodeToString(parentMsg.Message.Params))
+
+		for _, child := range util.GetChildMessagesOf(parentMsg) {
+			toCode, _ := getActorCode(ctx, child.Message.To)
+
+			toActorCode := "<Unknown>"
+			if !toCode.Equals(cid.Undef) {
+				toActorCode = toCode.String()
+			}
+			fromEthAddress := getEthAddress(child.Message.From)
+			toEthAddress := getEthAddress(child.Message.To)
+
+			vmMsg := &fevm.FEVMVMMessage{
+				Height:          int64(parentMsg.Height),
+				TransactionHash: transaction.Hash.String(),
+				StateRoot:       parentMsg.StateRoot.String(),
+				Source:          parentMsg.Cid.String(),
+				Cid:             getMessageTraceCid(child.Message).String(),
+				To:              child.Message.To.String(),
+				From:            child.Message.From.String(),
+				FromEthAddress:  fromEthAddress,
+				ToEthAddress:    toEthAddress,
+				Value:           child.Message.Value.String(),
+				GasUsed:         0,
+				ExitCode:        int64(child.Receipt.ExitCode),
+				ActorCode:       toActorCode,
+				Method:          uint64(child.Message.Method),
+				Index:           child.Index,
+				Params:          base64.StdEncoding.EncodeToString(child.Message.Params),
```

Comment:
Referring to the params value returned by `lily chain getmessage`, we also encode params value in Base64.

Reply:
what's the benefit to encode in base64?

Reply:
I think the one of reasons for encoding `params` into base64 is to ensure compatibility: Base64 encoding is a widely supported format, it is able to to mitigate risks across different operating systems and databases.